test/generics.jl

```diff
@@ -22,3 +22,8 @@ for i in 1:length(pmats)
     @test Matrix(3 * pmats[i])   == Matrix(pmatsa[i])
     @test Matrix(pmats[i] * 3)   == Matrix(pmatsa[i])
 end
+
+# issue #123
+@test isposdef(PDMat([1.0 0.0; 0.0 1.0]))
+@test isposdef(PDiagMat([1.0, 1.0]))
+@test isposdef(ScalMat(2, 3.0))
```

Comment:
Can you add the same trivial tests for `ishermitian`?

Reply:
Only `isposdef` was broken, so I only added tests for these failing examples. `ishermitian` works fine for `PDMat`, `PDiagMat`, and `ScalMat` already without this PR, it is just more efficient since it does not have to check all entries explicitly with this PR. So the actual change in this PR can't be tested.